lib/src/kbasesearchengine/events/handler/EventHandler.java

```diff
@@ -87,4 +89,15 @@ SourceData load(List<GUID> guids, Path file)
      * @return true if the event is expandable, false otherwise.
      */
     boolean isExpandable(StoredStatusEvent parentEvent);
+
+    /** Takes the specified event and returns a new event object that is a clone
+     * of the specified event and that reflects the latest state of the object
+     * that it is an event for.
+     *
+     * @param ev event to update
+     * @return a new updated event
+     * @throws UnprocessableEventIndexingException
+     */
+    StatusEvent updateEvent(StatusEvent ev)
```

Comment:
Maybe name updateObjectEvent to be clear

Reply:
This is a concrete detail and should be left out of the interface IMO.

Reply:
But in that case the current implementation is violating the interface because it's not updating a certain set of events.

I'm not advocating that it should update those events since there's currently no need to do so, but in that case the interface needs to be more granular to allow only implementing the desired behavior.